pkg/master/controller/crdregistration/crdregistration_controller.go

```diff
@@ -117,22 +120,16 @@ func (c *crdRegistrationController) Run(threadiness int, stopCh <-chan struct{})
 		return
 	}
 
-	// process each item in the list once
-	if crds, err := c.crdLister.List(labels.Everything()); err != nil {
-		utilruntime.HandleError(err)
-	} else {
-		for _, crd := range crds {
-			for _, version := range crd.Spec.Versions {
-				if err := c.syncHandler(schema.GroupVersion{Group: crd.Spec.Group, Version: version.Name}); err != nil {
-					utilruntime.HandleError(err)
-				}
-			}
-		}
-	}
-	close(c.syncedInitialSet)
+	// At the point, registered event handlers are already delivered.
+	// Informer framework will first deliver an OnAdd() event handler for each
+	// object and only then will start sending the stream of incoming events.
+	// We start a single work to process those initial handlers and wait until
+	// all of them are processed.
+	go wait.Until(c.runWorker, time.Second, stopCh)
+	c.WaitForInitialSync()
```

Comment:
This doesn't yet work, because processing happens in a different thread. I know how to do it, and will get back to it in the next couple days.